docs/src/main/asciidoc/kafka-streams.adoc

```diff
@@ -322,191 +468,343 @@ public static class WordCountProcessorApplication {
 
-Here is the property to set the contentType on the inbound.
+By default, Spring Cloud Stream will use `application/json` as the content type and use an appropriate json message converter.
+You can use custom message converters by using the following property.
+```
+spring.cloud.stream.bindings.process_output.contentType
+```
 
-[source]
-----
-spring.cloud.stream.bindings.input.contentType: application/json
-----
+When native encoding/decoding is disabled, binder will not do any inference as in the case of native Serdes.
+Applications need to explicitly provide all the configuration options.
+For that reason, it is generally advised to stay with the default options for de/serialization and stick with native de/serialization provided by Kafka Streams when you write Spring Cloud Stream Kafka Streams applications.
+The one scenario in which you must use message conversion capabilities provided by the framework is when your upstream producer is using a specific serialization strategy.
+In that case, you want to use a matching deserialization strategy as native mechanisms may fail.
 
-Here is the property to enable native decoding.
+=== Configuration Options
 
-[source]
-----
-spring.cloud.stream.bindings.input.nativeDecoding: true
-----
+This section contains the configuration options used by the Kafka Streams binder.
+
+For common configuration options and properties pertaining to binder, refer to the <<binding-properties,core documentation>>.
+
+==== Kafka Streams Properties
+
+The following properties are available at the binder level and must be prefixed with `spring.cloud.stream.kafka.streams.binder.`
 
-If native decoding is enabled on the input binding (user has to enable it as above explicitly), then the framework will
-skip doing any message conversion on the inbound. In that case, it will switch to the SerDe set by the user. The `valueSerde`
-property set on the actual output binding will be used. Here is an example.
+configuration::
+  Map with a key/value pair containing properties pertaining to Apache Kafka Streams API.
+  This property must be prefixed with `spring.cloud.stream.kafka.streams.binder.`.
+ Following are some examples of using this property.
 
 [source]
 ----
-spring.cloud.stream.kafka.streams.bindings.input.consumer.valueSerde: org.apache.kafka.common.serialization.Serdes$StringSerde
+spring.cloud.stream.kafka.streams.binder.configuration.default.key.serde=org.apache.kafka.common.serialization.Serdes$StringSerde
+spring.cloud.stream.kafka.streams.binder.configuration.default.value.serde=org.apache.kafka.common.serialization.Serdes$StringSerde
+spring.cloud.stream.kafka.streams.binder.configuration.commit.interval.ms=1000
 ----
 
-If this property is not set, it will use the default SerDe: `spring.cloud.stream.kafka.streams.binder.configuration.default.value.serde`.
+For more information about all the properties that may go into streams configuration, see StreamsConfig JavaDocs in
+Apache Kafka Streams docs.
+
+brokers::
```

Comment:
Does this section require formatting with code blocks?

Reply:
hmm - not sure. I didn't change this section much from what was there previously.

Reply:
No, I was wrong. I see them rendering correctly in the generated docs.